### PR TITLE
[Sprint 6] Feature Logic Accept And Reject Order In Order Management Shop

### PR DESCRIPTION
**Pull request by Tuanlvt**
[Sprint 6] Feature Logic Accept And Reject Order In Order Management Shop
Link redmine : https://dev.framgia.com/redmine/issues/120156
Task : Shop Information
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/data/source/remote/api/service/FOrderApi.java [hunk 1]
```diff
@@ -134,8 +135,8 @@
             @Body OrderRequest orderRequest);
 
     @PUT("v1/dashboard/order_products/0")
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/remote/api/service/FOrderApi.java]: ?``/0``
Please declare with me why put ``/0`` in here

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopAdapter.java [hunk 2]
```diff
@@ -106,4 +113,8 @@ public void updateData(List<Order> orders) {
         mOrders.addAll(orders);
         notifyDataSetChanged();
     }
+
+    public interface OrderManagementListener {
+        void onAcceptAndRejectOrderManager(OrderManagement acceptAndRejectOrdermanagementRequest);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopAdapter.java]: rename  method ``onAcceptAndRejectOrderManager`` -> ``onAcceptOrRejectOrderManager``

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 3]
```diff
@@ -18,6 +19,14 @@
         void onGetListOrderManagementShopSuccess(List<Order> orders);
 
         void onGetListOrderManagementShopError(BaseException exception);
+
+        void onAcceptAndRejectOrderSuccess();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: like above

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 4]
```diff
@@ -18,6 +19,14 @@
...
+
+        void onAcceptAndRejectOrderError(Exception exception);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: like above

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 5]
```diff
@@ -18,6 +19,14 @@
...
+
+        void onDoneOrderSuccess(List<Order> orders);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: ``onDoneOrderSuccess`` -> ``onOrderSuccess``

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 6]
```diff
@@ -18,6 +19,14 @@
...
+
+        void onDoneOrderError(Exception exception);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: like above

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 7]
```diff
@@ -26,5 +35,10 @@
     interface Presenter extends BasePresenter {
 
         void onGetListOrderManagementShop(int shopId);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: Please rename methods in presenter
Presenter contain action 
so you must remove prefix ``on``

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 8]
```diff
@@ -26,5 +35,10 @@
...
+
+        void onAcceptAndRejectOrder(int shopId, OrderManagement acceptAndRejectInOrder);
+
+        void onNotifyDoneOrderToServer(int shopId);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: like above

File: app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java [hunk 9]
```diff
@@ -26,5 +35,10 @@
     interface Presenter extends BasePresenter {
 
         void onGetListOrderManagementShop(int shopId);
+
+        void onAcceptAndRejectOrder(int shopId, OrderManagement acceptAndRejectInOrder);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/ordershop/OrderShopContract.java]: like above

File: app/src/main/java/com/framgia/forder/data/source/remote/api/service/FOrderApi.java (hunk 1 above)
Comment (Tuanlvt) [inline app/src/main/java/com/framgia/forder/data/source/remote/api/service/FOrderApi.java]: cái đó là bên viết APiI . nó bắt buộc phải để số thể nó mới cho gọi đến hàm PUT anh.

Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/remote/api/service/FOrderApi.java]: OK

**Pull request merged**